reference sheets (#29)
SciBorgs/SciGuides: reference-sheets/Simulation.md

# Robot Simulation

> Simulation in WPILib allows for code and logic to be tested onboard your computer, rather than physical hardware. It's exceptionally useful for when construction and electronics have not finished the robot, but you have code to run!

Reviewer: ```suggestion
Simulation in WPILib allows for code and logic to be tested onboard your computer, rather than with physical hardware. It's exceptionally useful for when mechanical hasn't finished the robot, but you have code to run!
```

There are a few different facets of simulation to take note of before you can start, including:

- [The Simulation GUI](https://docs.wpilib.org/en/stable/docs/software/wpilib-tools/robot-simulation)
- [WPILib's simulation classes](#simulation-classes)
  - [Physics simulators](https://docs.wpilib.org/en/stable/docs/software/wpilib-tools/robot-simulation/physics-sim.html)
  - [Mechanism2d](https://docs.wpilib.org/en/stable/docs/software/dashboards/glass/mech2d-widget.html)
  - [Field2d](https://docs.wpilib.org/en/stable/docs/software/dashboards/glass/field2d-widget.html)
- [Logging & Dashboards](#logging--dashboards)

## Logging & Dashboards

> In testing, it is common to want to directly observe robot measurements and values in order to tune your systems or debug. This is also incredibly important when working with simulation, as you otherwise have no reference to what is going on without a physical robot.

Reviewer: talk about viewing past logs as well -- that's as important as looking at data while you're testing
